pkg/providers/aws/provider_redis.go

```diff
@@ -99,6 +101,34 @@ func (p *AWSRedisProvider) CreateRedis(ctx context.Context, r *v1alpha1.Redis) (
 	// setup aws elasticache cluster sdk session
 	cacheSvc, stsSvc := createAWSService(stratCfg, providerCreds)
 
+	// Retrieve service maintenance updates, create and export Prometheus metrics
+	output, err := cacheSvc.DescribeServiceUpdates(&elasticache.DescribeServiceUpdatesInput{})
+	if err != nil {
+		logrus.Info(fmt.Sprintf("elasticache serviceupdates error: %s:", err))
+	} else {
+		logrus.Info(fmt.Sprintf("there are elasticache serviceupdates: %s available", len(output.ServiceUpdates)))
+		metricName := "cro_aws_elasticache_service_maintenance"
+		for _, su := range output.ServiceUpdates {
+			metricLabels := map[string]string{}
+
+			metricLabels["AutoUpdateAfterRecommendedApplyByDate"] = strconv.FormatBool(*su.AutoUpdateAfterRecommendedApplyByDate)
```

Comment:
Do you know if any of these labels can be `nil` or are they always included in `DescribeServiceUpdatesOutput`?

Reply:
That I don't know will need to look at the SDK docs

Reply:
Yeah it is worth just to clarify, to avoid any unwanted behavior

Reply:
From the AWS Documentation it does not mention that these results contain optional data, from this we're going to assume they will always be returned.